### DISCUSSION TITLE
Ability to define responsive styles in theme

### DISCUSSION THREAD

**Discussion by subodhpareek18** [mantinedev/mantine#3200] [Feature requests]
The ability to provide default props and default styles to components in the `<MantineProvider />` is really helpful. And the official way to do responsive styles without involving runtime JS would be to use `createStyles` or the `sx` props which is also great.

However if I wanted to not have to write my thin responsive wrapper on top of each component it would be a game-changer DX in my opinion.

If there was a way to write styles like so (or a different API even):
```js
<MantineProvider
   theme={{
    components: {
      Text: { styles: { fontSize: 18, [someMediaQuery]: { fontSize: 16 } } },
      Title: { styles: { fontSize: 44, [someMediaQuery]: { fontSize: 32 } } },
      Container: { styles: { padding: 90, [someMediaQuery]: { fontSize: 48 } } },
      Button: { styles: { padding: 16, [someMediaQuery]: { padding: 24 } } },
    }
  }}
>
{/* */}
</MantineProvider>
```

This would mean, I can define a bunch of my primary theme and responsive props and styles up top and the rest of my code can just grab the components directly from mantine. 

```js
<Container>
  <Title />
  <Text />
  <Button />
   {/* */}
</Container>
```

As opposed to writing code like:

```js

export const WrappedText = ({ children, ...props }: TextProps) => {
  <Text {...props} sx={{ // resonsive code here }} >
   {children}
  </Text> 
}

// bunch more components
```

```js
<WrappedContainer>
  <WrappedTitle />
  <WrappedText />
  <WrappedButton />
   {/* */}
</WrappedContainer>
```

It is still possible mind you, by using the `useMediaQuery` while providing props to the theme, but that involves JS to run before the CSS can be updated according to the screen, leading to horrible CLS and main thread being blocked. Which leads to poor UX, and performance and pagespeed scores.

```js
const sm = useMediaQuery('max-width(something)`)
```
```js
<MantineProvider
   theme={{
    components: {
      Text: { styles: { fontSize: sm ? 16 : 18 } },
      Title: { styles: { fontSize: sm ? 32 : 44 } },
      Container: { styles: { padding: sm ? 48 : 90 } },
      Button: { styles: { padding: sm ? 16 : 24 } },
    }
  }}
>
{/* */}
</MantineProvider>
```

I know there are some other changes being planned for v6, but I'm not sure if it's also planned to support them at the global theme level. So in case it isn't, I just wanted to share my opinion.

-----

The library is super amazing btw, haven't had such fun writing react ui code in some time! 
So many thanks 🎉

**Comment by rtivital** [comment]
You can add any styles (including responsive) using MantineProvider – https://mantine.dev/theming/mantine-provider/#styles-on-mantineprovider

**Reply by subodhpareek18** [reply]
Oh, that's great! Missed it entirely.

Would be helpful if this is documented with an example on the Responsive Styles section. Would be happy to make a PR if you also think the same.
